lite2/client.go

```diff
@@ -442,3 +613,62 @@ func (c *Client) compareNewHeaderWithRandomAlternative(h *types.SignedHeader) er
 
 	return nil
 }
+
+func (c *Client) removeNoLongerTrustedHeadersRoutine() {
+	ticker := time.NewTicker(c.removeNoLongerTrustedHeadersPeriod)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ticker.C:
+			c.RemoveNoLongerTrustedHeaders(time.Now())
+		case <-c.quit:
+			return
+		}
+	}
+}
+
+// RemoveNoLongerTrustedHeaders removes no longer trusted headers (due to
+// expiration).
+//
+// Exposed for testing.
+func (c *Client) RemoveNoLongerTrustedHeaders(now time.Time) {
+	// 1) Get the oldest height.
+	oldestHeight, err := c.trustedStore.FirstSignedHeaderHeight()
+	if err != nil {
+		c.logger.Error("can't get first trusted height", "err", err)
+		return
+	}
+
+	// 2) Get the latest height.
+	latestHeight, err := c.LastTrustedHeight()
+	if err != nil {
+		c.logger.Error("can't get last trusted height", "err", err)
+		return
+	}
+
+	// 3) Remove all headers (except the very last one) that are outside of
+	// the trusting period.
+	for height := oldestHeight; height < latestHeight; height++ {
+		h, err := c.trustedStore.SignedHeader(height)
+		if err != nil {
+			c.logger.Error("can't get a trusted header", "err", err, "height", height)
+			continue
```

Comment:
Should we panic? When would this happen?

Reply:
> When would this happen?

problem with underlying database. no panics please. header will be removed during next iteration.